app/user.go

```diff
@@ -318,9 +318,30 @@ func (a *App) createUserOrGuest(c request.CTX, user *model.User, guest bool) (*m
 		})
 	}
 
+	_, cwsErr := a.SendSubscriptionHistoryEvent(ruser.Id)
+	if cwsErr != nil {
+		c.Logger().Error("Failed to create/update the SubscriptionHistoryEvent", mlog.Err(cwsErr))
+	}
+
 	return ruser, nil
 }
 
+func (a *App) SendSubscriptionHistoryEvent(userID string) (*model.SubscriptionHistory, error) {
+	license := a.Srv().License()
+
+	// No need to create a Subscription History Event if the license isn't cloud
+	if !license.IsCloud() {
+		return nil, nil
+	}
+
+	// Get user count
+	userCount, err := a.Srv().Store().User().Count(model.UserCountOptions{})
```

Comment:
More just a note but we might have to keep an eye on replication lag here. The user is created before this function is called but the count function grabs from the read replica so it's possible the user count could not include this new user